lua/caching_handlers.lua

```diff
@@ -110,9 +110,10 @@ function caching_handlers._caching_handler(request_info, cacheability_info)
         ngx.var.request_uri,
         ngx.req.get_headers()
     )
-
-    local headers = response.uncacheable_headers
     local post_request
+    local headers = response.uncacheable_headers
+    headers[spectre_common.HEADERS.ORIGINAL_STATUS] = response.no_response and -1 or response.status
```

Comment:
i'm trying to figure out if this `-1` thing is going to bite us later... I actually don't understand why we'd ever have `no_response=true`, since the service mesh itself should be returning status codes even when spectre can't reach any upstream. maybe connection timeout, but that's really still a 503 probably. This might mean more of like, the HTTP connection to the localhost envoy went wrong. I'd kind of be on board logging which condition in particular led to this, in addition to setting this to -1 here, which lumps a few things together.

I see there is already code to emit metrics to `spectre.errors` when this happens, and poking in Signalfx quickly I see there are never any of these metrics present from this code path in prod over the past 30 days, so this checks out with my mental model, and I don't think we'll ever see these in practice.
`B = data('spectre.errors', filter=filter('ecosystem', 'prod') and filter('instance_name', 'main')).publish(label='B')` only shows status_codes: -1 (cassandra error / timeout) and 404. (both of these metrics are coming from different code paths to the same metric name)

Reply:
@thebostik 
You're right, `-1` here is mostly an indicator of misconfigured or non-running Envoy (eg. not listen on a port OR unix socket). 
Connections timeouts ideally should be handled by the Envoy and we should get 503 from it. 
Probably we even should not have any timeouts set on the casper side and instead terminate current Lua thread on upstream connection close (that would be a timeout from the upstream/client side). Not sure is it possible or not in openresty.